api/extension-guides/chat.md

```diff
@@ -0,0 +1,204 @@
+
+![Chat concepts explanation](images/chat/chat.png)
+
+This screenshot shows the following chat concepts with a sample extension:
+1. `@cat` explicitly being called by the user via the `@` syntax
+2. Command explicitly called via the `/` syntax
+3. User query
+4. Copilot using the contributed `@cat` participant
+5. Markdown response that is fully controlled by the contributed `@cat` participant
+6. Code part of the markdown response
+7. [Button response](#response-stream)
+8. [Follow-ups](#follow-ups)
+9. Chat input for the user to continue. `description` of `@cat` is used as a placeholder
+
+
+## Chat Extension
+
+The [chat sample](https://github.com/microsoft/vscode-extension-samples/tree/main/chat-agent-sample) can be used as a starting point for your chat extension. It is a simple cat tutor that explains computer science topics using cat metaphors.
+
+![Diagram showing how extension can contribute to chat](images/chat/diagram.png)
+
+To start and register a chat participant you can do the following:
+
+```typescript
+const cat = vscode.chat.createChatParticipant('cat', handler);
+```
+
+> **Note:**: The Chat and Language Model API is in a [proposed state](https://code.visualstudio.com/api/advanced-topics/using-proposed-api) and we are actively working on adding more functionality. Share your feedback in [this GitHub issue](https://github.com/microsoft/vscode/issues/199908) or create new issues.
+
+### Commands
+
+Chat participants can contribute `/commands`, which are shortcuts to specific functionality provided by the extension. One of the tasks when answering questions is to determine the intent, understanding what you want to do. VS Code can infer that "Create a new workspace with Node.js Express Pug TypeScript" means that you want a new project, but "`@workspace /new` Node.js Express Pug TypeScript" is explicit, concise, and saves typing time. By pressing `/` VS Code will offer a list of registered commands with their description.
+
+![List of commands in chat for @workspace](images/chat/commands.png)
+
+Chat participants can contribute commands with their description using the `commandProvider`:
+
+```typescript
+cat.commandProvider = {
+    provideCommands(token) {
+        return [
+            { name: 'teach', description: 'Pick at random a computer science concept then explain it in purfect way of a cat' },
+            { name: 'play', description: 'Do whatever you want, you are a cat after all' }
+        ];
+    }
+};
+```
+
+
+### Response Stream
+
+Using a response stream extensions can respond to user queries with different content types: markdown, images, references, buttons and file trees. For example to generate this response:
+
+![Response from the cat extension that includes code, markdown and a button](images/chat/stream.png)
+
+
+An extension can use the response stream in the following way:
+
+```typescript
+stream.markdown('```typescript \n')
```

Comment:
I think it would be better to show that you don't _have_ to call this with one line of markdown at a time

Reply:
Ok. Moved all into one markdown call